simple_history/models.py

```diff
@@ -645,6 +646,17 @@ def post_delete(self, instance, using=None, **kwargs):
         else:
             self.create_historical_record(instance, "-", using=using)
 
+    def pre_delete(self, instance, **kwargs):
+        """
+        pre_delete method to ensure all deferred fileds are loaded on the model
+        """
+        fields = self.fields_included(instance)
+        field_attrs = {field.attname for field in fields}
+        deferred_attrs = instance.get_deferred_fields()
+        # Load all deferred fields that are present in fields_included
+        for attr in field_attrs.intersection(deferred_attrs):
```

Comment:
Maybe it would be better to load all deferred fields in a single query, as Django loads fields one at a time on access (See https://docs.djangoproject.com/en/5.1/ref/models/querysets/#defer)

Updated code may look something like this:
```
instance.refresh_from_db(fields=field_attrs.intersection(deferred_attrs))
```

Reply:
Yes, good idea,  I have implemented this.  Hopefully this gets reviewed and accepted soon!